File: backend/src/tact/schemas/context.py
from datetime import datetime

from pydantic import BaseModel


class ContextCreate(BaseModel):
    content: str


class ContextUpdate(BaseModel):
    content: str


class ContextResponse(BaseModel):
    id: str
    project_id: str | None
    time_code_id: str | None
    content: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: webui/model/context.go
package model

type ContextDocument struct {
	ID         string  `json:"id"`
	ProjectID  *string `json:"project_id"`
	TimeCodeID *string `json:"time_code_id"`
	Content    string  `json:"content"`
	CreatedAt  Time    `json:"created_at"`
	UpdatedAt  Time    `json:"updated_at"`
}
